services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocumentTest.java

```diff
@@ -304,6 +305,34 @@ void access_CustomType_without_AttributeConverterProvider() {
         assertThat(docWithCustomProvider.get("customMapValue", EnhancedType.of(CustomClassForDocumentAPI.class))).isNotNull();
     }
 
+    @Test
+    void error_When_listUsed() {
```

Comment:
Nit: The test case names in this class don't seem to follow a rule, but can we use something like
`listUsed_missingObjectConverter_throwsIllegalStateException`?